### ISSUE TITLE
Recommendation for service.name to be autodetected when not manually set

### ISSUE THREAD

**Issue by Kronos11** [open-telemetry/opentelemetry-specification#3210]
**What are you trying to achieve?**
I believe changing service.name to Recommended instead of Required will allow SDKs to properly set the values when not found to an appropriate default instead of (unknown_service). This is a terrible default for many cases. One in particular is an AspNet Framework application that uses (see: https://github.com/open-telemetry/opentelemetry-dotnet/issues/2781).
There are many other cases where there is not an appropriate way to set the service.name at a feasible  default when it is not specified or specified ONLY by an Environment variable.

What did you expect to see?
service.name should be set as Recommended instead of Required. This will allow SDK libraries to set defaults when the user hasn't specified them, or they are unable to specify them in certain cases.

**Additional context.**
Other tickets that have problems when service.name should not be required
[1](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2189)
[2](https://github.com/open-telemetry/opentelemetry-dotnet/issues/2027)
[3](https://github.com/open-telemetry/opentelemetry-specification/issues/1497)
[4](https://github.com/open-telemetry/opentelemetry-specification/issues/3025)

See an appropriate default set [here ](https://github.com/signalfx/splunk-otel-dotnet/blob/main/src/Splunk.OpenTelemetry.AutoInstrumentation/Helpers/ServiceNameHelper.cs) by the splunk library.

**Comment by Kronos11** [comment]
After reviewing previous [issues ](https://github.com/open-telemetry/opentelemetry-specification/pull/1269)I see why this decision was made. I do wonder if it could instead be up to the auto instrumentor to decide if they want to provide a better default value for the user in case it's needed as in my case.

The other option I suppose would be for the user to decide if they really do want a fallback or default as part of configuration.
What do you think?

**Comment by pellared** [comment]
I agree with issue and I do not think that it is only applicable for ASP.NET. It is not rare the instrumented application works on top some "hosting process" such us `dotnet`, `java`, `python3`, `w3wp`.

However, some ecosystems could bring some useful defaults. This would be extremely useful especially for auto-instrumentations.

> If the value was not specified, SDKs MUST fallback to `unknown_service:` concatenated with `process.executable.name`, e.g. `unknown_service:bash`

I propose a change in the specification to allow different values than `process.executable.name` such as "name of the main program entry" (which following for .NET it can be the name of the executing assembly, for Python it could be the name of the main module, for Java - the main package name). 

Would such change in the specification be seen as backwards compatible? The `process.executable.name` would still be allowed.

**Comment by arminru** [comment]
From the spec triage meeting - @carlosalberto will follow up

**Comment by Aneurysm9** [comment]
> I think that the spec allows using additional "resource detectors". I guess we can create/use more detectors that would set the `service.name`.
> 
> Am I correct?

This is how I view it.  I don't think that any change is required unless we want to clarify that users can use resource detectors to detect any resource attribute.  The requirement that the SDK provide an ultimate fallback is only there because the `service.name` resource attribute is mandatory and something needs to ensure that it exists.  In the Go SDK we do this [with a resource detector](https://github.com/open-telemetry/opentelemetry-go/blob/3015c86cfd1e1868b9c57a998d44ebe25acffedc/sdk/resource/resource.go#L218).

**Comment by pellared** [comment]
**From 07-2023-03-07 SIG meeting** 

It is acceptable to use resource detectors which is already done e.g. by [Java](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/resource-providers/src/main/java/io/opentelemetry/contrib/resourceproviders) . The SDKs should apply the the resource detectors before falling to the `unknown_service` service name.

There was no consensus if it needs to be explicitly defined in the specification.

**Comment by pellared** [comment]
@open-telemetry/specs-approvers I think this issue can be closed.
